dali/test/python/test_operator_normalize.py

```diff
@@ -0,0 +1,255 @@
+        scalar_stddev = self.scalar_stddev(data)
+        ext_mean = self.normalize(data, mean = mean)
+        ext_stddev = self.normalize(data, stddev = stddev)
+        ext_all = self.normalize(data, mean = mean, stddev = stddev)
+        scalar_mean_ext = self.scalar_mean(data, stddev = stddev)
+        scalar_stddev_ext = self.scalar_stddev(data, mean = mean)
+        if self.axes is None:
+            scalar_params = self.scalar_params(data)
+
+        out = [data, mean, stddev, normalized, scalar_mean, scalar_stddev,
+                ext_mean, ext_stddev, ext_all, scalar_mean_ext, scalar_stddev_ext]
+        if self.axes is None:
+            out.append(scalar_params)
+        return out
+
+    def check_batch(self, data, mean, stddev, normalized, scalar_mean, scalar_stddev,
+                ext_mean, ext_stddev, ext_all, scalar_mean_ext, scalar_stddev_ext, scalar_params = None):
+        axes = self.axes
+        if type(axes) is list:
+            axes = tuple(axes)
+        batch = self.batch
+        mean_func = custom_mean(batch, axes)
+        stddev_func = custom_stddev(batch, axes)
+
+        ref = normalize_list(batch, data, axes)
+        ref_scalar_mean = normalize_list(batch, data, axes, mean = 1)
+        ref_scalar_stddev = normalize_list(batch, data, axes, stddev = 2)
+        mean, = mean_func(data)
+        stddev, = stddev_func(data)
+        ref_ext_mean = normalize_list(batch, data, axes, mean = mean)
+        ref_ext_stddev = normalize_list(batch, data, axes, stddev = stddev)
+        ref_ext_all = normalize_list(batch, data, axes, mean = mean, stddev = stddev)
+        ref_ext_all = normalize_list(batch, data, axes, mean = mean, stddev = stddev)
+        ref_scalar_mean_ext = normalize_list(batch, data, axes, mean = 1, stddev = stddev)
+        ref_scalar_stddev_ext = normalize_list(batch, data, axes, mean = mean, stddev = 2)
+        check(scalar_stddev, ref_scalar_stddev)
+        check(scalar_mean, ref_scalar_mean)
+        check(ext_mean, ref_ext_mean)
+        check(ext_stddev, ref_ext_stddev)
+        check(ext_all, ref_ext_all)
+        check(scalar_mean_ext, ref_scalar_mean_ext)
+        check(scalar_stddev_ext, ref_scalar_stddev_ext)
+        if scalar_params is not None:
+            ref_scalar_params = normalize_list(batch, data, axes, mean = 1, stddev = 2)
+            check(scalar_params, ref_scalar_params)
+
+    def iter_setup(self):
+        self.feed_input(self.input_data, generate_data(self.dims, self.batch_size, self.batch, self.axes))
+
+def to_list(tensor_list):
+    out = []
+    for i in range(len(tensor_list)):
+        out.append(tensor_list.at(i))
+    return out
+
+np.random.seed(seed=1337)
+
+def mask2axes(mask):
+    out = []
+    a = 0
+    while mask:
+        if mask & 1:
+            out.append(a)
+        mask >>= 1
+        a += 1
+    return out
+
+def all_axes(dim):
+    yield None
+    for mask in range(1, 1 << dim):
+        yield mask2axes(mask)
+
+def main():
```

Comment:
Does nose runs main? I think it looks for *test* functions and tries to run them.